File: BOJ/Recommended1/Bronze/16283/16283.py
a,b,n,w = map(int, input().split())
arr = []
for x in range(1,1000) :
	if n-x >= 1 and a*x + b*(n-x) == w : arr.append(f"{x} {n-x}")
	if len(arr) > 1 : break
if len(arr) > 1 or len(arr) == 0 : print(-1)
else : print(arr[0])

File: BOJ/Recommended1/Bronze/16283/16283.swift
let input = readLine()!.split(separator : " ").map{Int(String($0))!}
var arr : [String] = []
for x in 1..<1000 {
	if input[2]-x >= 1 && input[0]*x + input[1]*(input[2]-x) == input[3] {arr.append("\(x) \(input[2]-x)")}
	if arr.count > 1 {break}
}
if arr.count > 1 || arr.count == 0 {print(-1)}
else {print(arr[0])}